_content/tour/ukr/pointers.article

```diff
@@ -1,151 +1,144 @@
+    inc2:   Значення[ 0xc000050738 ] Адреса[ 0xc000050748 ] Вказує на [ 11 ]
+    count:  Значення[ 11 ]  Адреса[ 0xc000050738 ]
 
-** Notes
+** Примітки
 
-- Use pointers to share data.
-- Values in Go are always pass by value.
-- "Value of", what's in the box. "Address of" ( & ), where is the box.
-- The ( * ) operator declares a pointer variable and the "Value that the pointer points to".
+- Використовуй вказівники, щоб ділитися даними.
+- Значення в Go завжди передаються за значенням.
+- "Значення", це те, що в коробці. "Адреса" ( & ), значить де ця коробка.
+- Оператор ( * ) визначає змінну-вказівник і "Значення, куди цей вказівник вказує".
 
-** Escape Analysis
+** Аналіз втечі (escape analysis)
 
-The algorithm the compiler uses to determine if a value should be constructed
-on the stack or heap is called "escape analysis". The name of the algorithm makes
-it sound like values are constructed on the stack first and then escape (or move)
-to the heap when necessary. This is NOT the case. The construction of a value only
-happens once, and the escape analysis algorithm decides where that will be (stack
-or heap). Only construction on the heap is called an allocation in Go.
+Алгоритм компілятора, щоб визначити, чи треба змінну створювати у стеку чи кучі,
+називається "аналіз втечі" (escape analysis). Імʼя алгоритма звучить так,
+ніби змінні спочатку створюються у стеку і потім втікають у кучу, коли це потрібно.  
+Це НЕ ПРАВДА. Створення значення відбувається тільки один раз
+і алгоритм аналізу втечі визначає де ця зміна буде створена (стек або куча).
+Тільки створення змінної у кучі називається виділенням памʼяті (allocation) у Go.
 
-Understanding escape analysis is about understanding value ownership. The idea is,
-when a value is constructed within the scope of a function, then that function
-owns the value. From there ask the question, does the value being constructed still
-have to exist when the owning function returns? If the answer is no, the value can
-be constructed on the stack. If the answer is yes, the value must be constructed on
-the heap.
+Щоб зрозуміти аналіз втечі треба зрозуміти володіння значенням. Суть у тому,
+що якщо значення створюється в межах області видимості функції, тоді ця функція володіє значенням.
+Звідси постає питання, чи новостворене значення повинно існувати після завершення виконання функції?
+Якщо відповідь ні, то значення створюється у стеку. Якщо відповідь так, то значення створюється у кучі.
 
-Note: The ownership rule is a good base rule for identifying code that causes
-allocations. However, you must appreciate that escape analysis has flaws that can
-result in non-obvious allocations. Also, the algorithm takes opportunities to
-leverage compiler optimizations to save on allocations.
+Примітка: Правило володіння добре допомагає визначити код який провокує
+виділення памʼяті. Проте, тримайте в голові, що аналіз втечі містить свої недоліки, які можуть
```

Comment:
https://goroh.pp.ua/%D0%A1%D0%BB%D0%BE%D0%B2%D0%BE%D0%B2%D0%B6%D0%B8%D0%B2%D0%B0%D0%BD%D0%BD%D1%8F/%D1%82%D1%80%D0%B8%D0%BC%D0%B0%D1%82%D0%B8%20%D0%B2%20%D0%B3%D0%BE%D0%BB%D0%BE%D0%B2%D1%96

```suggestion
виділення памʼяті. Проте, майте на увазі, що аналіз втечі містить свої недоліки, які можуть
```